tests/integration/suite/daprd/metrics/actors/actors.go

```diff
@@ -0,0 +1,106 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actors
+
+import (
+	"context"
+	"net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/http/app"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(actors))
+}
+
+type actors struct {
+	place     *placement.Placement
+	scheduler *scheduler.Scheduler
+	triggered atomic.Int64
+
+	daprd *daprd.Daprd
+}
+
+func (a *actors) Setup(t *testing.T) []framework.Option {
+	a.scheduler = scheduler.New(t)
+
+	app := app.New(t,
+		app.WithHandlerFunc("/actors/myactortype/myactorid", func(http.ResponseWriter, *http.Request) {}),
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/remind/remindermethod", func(http.ResponseWriter, *http.Request) {
+			a.triggered.Add(1)
+		}),
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/foo", func(http.ResponseWriter, *http.Request) {}),
+		app.WithConfig(`{"entities": ["myactortype"]}`),
+	)
+
+	a.place = placement.New(t)
+	a.daprd = daprd.New(t,
+		daprd.WithConfigManifests(t, `
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+ name: schedulerreminders
+spec:
+ features:
+ - name: SchedulerReminders
+   enabled: true
+`),
+		daprd.WithInMemoryActorStateStore("mystore"),
+		daprd.WithPlacementAddresses(a.place.Address()),
+		daprd.WithSchedulerAddresses(a.scheduler.Address()),
+		daprd.WithAppPort(app.Port()),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(app, a.scheduler, a.place, a.daprd),
+	}
+}
+
+func (a *actors) Run(t *testing.T, ctx context.Context) {
+	a.scheduler.WaitUntilRunning(t, ctx)
+	a.place.WaitUntilRunning(t, ctx)
+	a.daprd.WaitUntilRunning(t, ctx)
+
+	grpcClient := a.daprd.GRPCClient(t, ctx)
+
+	_, err := grpcClient.RegisterActorReminder(ctx, &runtimev1pb.RegisterActorReminderRequest{
+		ActorType: "myactortype",
+		ActorId:   "myactorid",
+		Name:      "remindermethod",
+		DueTime:   "0s",
+		Period:    "1s",
+	})
+	require.NoError(t, err)
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.GreaterOrEqual(c, a.triggered.Load(), int64(1))
+	}, 30*time.Second, 10*time.Millisecond, "failed to wait for 'triggered' to be greater or equal 1, actual value %d", a.triggered.Load())
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		metrics := a.daprd.Metrics(c, ctx).All()
+		assert.GreaterOrEqual(c, a.triggered.Load(), int64(metrics["dapr_runtime_actor_reminders_fired_total"]))
```

Comment:
Shouldn't this be Equal?

Reply:
I did not set a repeat for the reminder, so I cannot guarantee the exact count, especially with CI timing variations, hence why I put the `GreaterOrEqual`. I can set a Repeat if desired, but we have other tests already ensuring correct trigger count at the same time.

Reply:
We do the same for [this other test with no repeats set](https://github.com/dapr/dapr/blob/release-1.15/tests/integration/suite/scheduler/metrics/actors.go#L118)

Reply:
Yes, however surely the metric number should be the same as the number of triggers?

Reply:
sure, I can change it for an additional check for count